salt/kosmtik.sls

```diff
@@ -0,0 +1,106 @@
+# Salt states for getting kosmtik installed for purposes of
+# improving the OpenStreetMap CartoCSS implementation and style.
+
+
+# A whole bunch of libraries that we'll need, plus a bunch
+# of extra fonts.
+# TODO: fill in exact requirements
+package-requirements:
+  pkg.installed:
+    - install_recommends: false
+    - pkgs:
+      - libgtk2.0-dev
+      - libwebkitgtk-dev
+      - protobuf-compiler
+      - libprotobuf-dev
+      - libgdal1-dev
+      - gdal-bin
+      - python-yaml
+      # here be the fonts
+      - ttf-dejavu
+      - ttf-unifont
+      - fonts-droid
+      - fonts-sipa-arundina
+      - fonts-sil-padauk
+      - fonts-khmeros
+
+# nodejs is what kosmtik runs on
+nodejs:
+  pkgrepo.managed:
+    - humanname: NodeSource node.js
+    - name: deb https://deb.nodesource.com/node/ trusty main
+    - dist: trusty
+    - file: /etc/apt/sources.list.d/nodesource.list
+    - key_url: https://deb.nodesource.com/gpgkey/nodesource.gpg.key
+  pkg.latest:
+    - pkgs:
+      - nodejs
+    - install_recommends: false
+    - refresh: true
+
+# Getting kosmtik from github, latest version
+kosmtik:
+  git.latest:
+    - rev: master
+    - name: https://github.com/kosmtik/kosmtik.git
+    - target: /usr/local/kosmtik
+    - depth: 1
+    - require:
+      - pkg: package-requirements
+  npm.bootstrap:
+    - name: /usr/local/kosmtik
+    - require:
+      - pkg: nodejs
+      - git: kosmtik
+  # localconfig.json -- can be edited, is stored in the home directory
+  file.managed:
+    - name: /root/localconfig.json
+    - contents: |
+        [
+          { 
+            "where": "Layer",
+            "if": {
+              "Datasource.type": "postgis"
```

Comment:
this will override **all** postgis layers, not just ones pulling from the gis db. You want to restrict it to just changing databases named gis, if you need to change it.